policy-controller/k8s/index/src/meshtls_authentication.rs

```diff
@@ -29,6 +29,12 @@ impl Spec {
                 let ns = tgt.namespace.as_deref().unwrap_or(&namespace);
                 let id = cluster.service_account_identity(ns, &tgt.name);
                 Ok(IdentityMatch::Exact(id))
+            } else if tgt.targets_kind::<Namespace>() {
+                let identity_suffix = format!(
+                    "{}.serviceaccount.identity.{}.{}",
+                    tgt.name, cluster.control_plane_ns, cluster.identity_domain
+                );
+                Ok(IdentityMatch::Suffix(vec![identity_suffix]))
```

Comment:
I'm a little surprised this works. The IdentityMatch::Suffix type holds a list of suffix components:

https://github.com/linkerd/linkerd2/blob/e962bf857c619c5e1988a1ccb11c2792e52e134b/policy-controller/core/src/identity_match.rs#L15-L31

The intent her is so that we explicitly differentiate exact string matches vs parsed suffix matches. I.e. a suffix like `bar.serviceaccount.linkerd.domain` should not match `foobar.serviceaccount.linkerd.domain`. So we tokenize the suffix and compare the segments individually.

Regardless, I'd prefer that we keep all of the identity formatting logic hung off of `ClusterInfo` so that it's all in one place.

https://github.com/linkerd/linkerd2/blob/c1a1430d1a524ff6a74ebf3a5127d276b5a3b26a/policy-controller/k8s/index/src/lib.rs#L67-L75

Why don't we:
* Add a `ClusterInfo::namespace_identity` helper
* Because `identity_domain` can include dots, it will probably need to format the string as `*.<ns>.serviceaccount.identity.<cpns>.<domain>`
* here, we use `ClusterInfo::namespace_identity(&tgt.name).parse::<IdentityMatch>()?`